test/common/testlib.py

```diff
@@ -1064,6 +1088,17 @@ def setUp(self):
         if opts.address and self.provision is not None:
             raise unittest.SkipTest("Cannot provision multiple machines if a specific machine address is specified")
 
+        if self._testlib__use_global_machine:
+            if self.provision is not None:
+                raise unittest.SkipTest("Cannot provision machines on a nonDestructive test")
```

Comment:
I'd switch the logic of the message and write "cannot mark tests non-destructive on classes that specify a provision".

Alternatively, just silently ignore it.  The non-destructive thing is really an optimisation at this point, isn't it?  We might also imagine a future where this *is* possible *and* meaningful: specifying the attribute on methods in a test class that specifies a provision would mean to reuse the provisioned machines.